### ISSUE TITLE
[feat] Standardize focus indicator for interactive components

### ISSUE THREAD

**Issue by marionnegp** [RedHat-UX/red-hat-design-system#276]
### Description

Some components, like CTAs, use different focus indicators than others, like jump links, for their focus states. These should be standardized as much as possible using the focus indicator from the design system kit.

### Suggested solution

The focus indicator is 2px thick. For an outer focus indicator, there should be a 1px margin between the focused element and the focus indicator.

![Screen Shot 2022-05-13 at 2 44 45 PM](https://user-images.githubusercontent.com/95588923/168347600-d4f92b22-b1b3-4cbc-a8bf-4e2922c804c2.png)
color: #06c 

![Screen Shot 2022-05-13 at 2 44 55 PM](https://user-images.githubusercontent.com/95588923/168347603-df2e405d-b39d-4f8a-bc32-d07a27ba60a6.png)
color: #73BCF7

![Screen Shot 2022-05-13 at 2 45 04 PM](https://user-images.githubusercontent.com/95588923/168347604-ab050df1-fc35-4d0e-b536-3720e2ef9347.png)
color: #fff

**Example:** This tertiary button uses the outer focus indicator.
![Screen Shot 2022-05-13 at 3 17 05 PM](https://user-images.githubusercontent.com/95588923/168374866-37a84f24-c200-457c-bb77-758b038983a9.png)

### Example API

```HTML
They should work similarly to the tabs and jump links under [Explore Red Hat Products](https://www.redhat.com/en#browse-products) on the homepage
```

### Additional context

Component focus states can be updated as they're found.

**Comment by bennypowers** [comment]
Does the [`outline`](https://developer.mozilla.org/en-US/docs/Web/CSS/outline) property suit your requirements or would you rather use `border`s? TL;DR: [border is more flexible but less performant](https://css-tricks.com/using-the-outline-property-as-a-collapsable-border/)

**Comment by marionnegp** [comment]
`Outline` sounds fine. `outline-offset' should also be an easier way to add space between the element and the focus indicator. 

Can outlines have rounded corners on more browsers now? I think we're actually using it for the jump link component's focus state, and that outline is slightly rounded.

<img width="206" alt="Screen Shot 2022-05-18 at 4 22 58 PM" src="https://user-images.githubusercontent.com/95588923/169149192-7141bce3-d94c-4650-ade7-dc4d4a73adc0.png">

**Comment by bennypowers** [comment]
Check out the last demo on https://developer.mozilla.org/en-US/docs/Web/CSS/outline

the outline will follow the border radius of the element on FF and Chrome (desktop mac), but not Safari

**Comment by marionnegp** [comment]
Cool, I don't think it's a big deal that border radius isn't supported for outlines on Safari. I think it's more important that we start getting focus states to look more similar. @coreyvickery or @ronniemcbride, chime in if you feel otherwise.
